server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -759,10 +779,57 @@ public int hashCode() {
             result = 31 * result + Long.hashCode(globalCheckpoint);
             result = 31 * result + Boolean.hashCode(inSync);
             result = 31 * result + Boolean.hashCode(tracked);
+            result = 31 * result + Boolean.hashCode(replicated);
             return result;
         }
     }
 
+    /**
+     * The type of replication used for inter-node replication.
+     */
+    public enum ReplicationMode implements Writeable {
```

Comment:
Should this class should be moved outside the tracker?

Reply:
Ack, will make the change.